### PR TITLE
Utils for analysis, classifier V0

### PR DESCRIPTION
**Pull request by aldokaraj**
- [x] Tool to manually label policies
- [x] Aded Dataset util in utils to support common operations: training, testing, subsetting etc
- [x] Trained a classifier with average precision around `93%`

### CODE REVIEW

File: privacy_bot/analysis/classifier.py [hunk 1]
```diff
@@ -1,3 +1,17 @@
+"""
+Privacy Bot - Text Classifier
+
+Usage:
+    classifier.py -p TRUE_POSITIVES  -n TRUE_NEGATIVES -e ENTITY
+
+Options:
+    -p --tp TRUE_POSITIVES        Local Path of TRUE_POSITIVES (tar file)
+    -n --tn TRUE_NEGATIVES        Local Path of TRUE_NEGATIVES (tar file)
+    -e --entity ENTITY            Two possible values: text, url
```
Comment (remusao) [inline privacy_bot/analysis/classifier.py]: Since we need one of them at least, we should probably use  an alternative in the `Usage` instead: `(--url | --text)`. Which would give:
```
Usage:
    classifier.py -p TRUE_POSITIVES  -n TRUE_NEGATIVES (--url | --text)
```

Also, you don't really need to specify a flag for the options `-p` and `-n` since they are always present anyway (it's also a matter of taste, I'm just proposing an alternative). Could be something like:
```
Usage:
    classifier.py <positive>  <negative> (--url | --text)
```

(They are now positional arguments)

File: privacy_bot/analysis/classifier.py [hunk 2]
```diff
@@ -19,25 +34,39 @@
...
+                    ngram_range=(3, 5)
+                )),
+                ('clf', SGDClassifier())
+])
+
 
 if __name__ == "__main__":
+    args = docopt(__doc__)
+    print(args)
+    print(args['--entity'])
```
Comment (remusao) [inline privacy_bot/analysis/classifier.py]: Is it for debugging?

File: privacy_bot/analysis/classifier.py [hunk 3]
```diff
@@ -19,25 +34,39 @@
...
     labelled_dataset = Dataset.tars_to_labelled(
-        true_positives_path='privacy_policy_positive.tar.bz2',
-        true_negatives_path='privacy_policy_negative.tar.bz2'
+        true_positives_path=args['--tp'],
+        true_negatives_path=args['--tn'],
+        entity=args['--entity']
     )
-    
+    #TODO: training_size, testing_size into args (when > labelled data)
     training, test = labelled_dataset.training_testing(
```
Comment (remusao) [inline privacy_bot/analysis/classifier.py]: What about specifying a percentage instead of number? Does not seem very handy if you change the size of the dataset. We could just wrap this http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html#sklearn.model_selection.train_test_split from scikit directly and offer the same arguments.

File: privacy_bot/analysis/utils.py [hunk 4]
```diff
@@ -8,6 +8,9 @@
 
 
 class Dataset():
```
Comment (remusao) [inline privacy_bot/analysis/utils.py]: Shouldn't `url dataset` and `policies dataset` be two different classes? Or at least be loaded by two different functions? It would look more 'scikit-like' to me. (judging by what they propose there: http://scikit-learn.org/stable/datasets/)

File: privacy_bot/analysis/utils.py [hunk 5]
```diff
@@ -40,19 +43,29 @@ def from_trained(path):
...
             names.append('policy')
 
         for policy in tn.query(lang=language):
-            data.append(policy.text)
+            if entity == 'text':
+                data.append(policy.text)
+            elif entity == 'url':
+                data.append(policy.url)
+            elif entity == 'html':
+                data.append(policy.html)
```
Comment (remusao) [inline privacy_bot/analysis/utils.py]: This seems to be an exact copy of the code, can we refactor it maybe?

File: privacy_bot/analysis/utils.py [hunk 6]
```diff
@@ -8,6 +8,9 @@
 
 
 class Dataset():
+    """
+    Primarily used to ease data analysis tasks (e.g. text classification)
+    """
     def __init__(self, data, names):
         self.data = data
         self.names = names
```
Comment (remusao) [inline privacy_bot/analysis/utils.py]: Aren't `self.names` and `self.labels` redundant? They seem to contain exactly the same information.

File: privacy_bot/analysis/utils.py [hunk 7]
```diff
@@ -71,6 +84,10 @@ def shuffle(self):
         return Dataset(data=data, names=names)
 
     def training_testing(self, training_size, testing_size):
+        """
+        Generates training and test from labelled data
+
+        """
```
Comment (remusao) [inline privacy_bot/analysis/utils.py]: This could probably be refactored to have less code copy.
